### ISSUE TITLE
[PROPOSAL] Possible improvement to show "caption" instead of "title" and "empty caption" fallback issue with TW lists

### ISSUE THREAD

**Issue by pmario** [Jermolene/TiddlyWiki5#6397]
## The Problem

From time to time we get issues by users that create "lists" in TW that contain empty elements. Those empty elements come from empty caption-fields. 

see: 

![grafik](https://user-images.githubusercontent.com/374655/149536437-4588f783-8764-40ce-b01f-48b0e6dabcee.png)

The core TOC macro has the same problem, which is even harder to see and debug: 

![grafik](https://user-images.githubusercontent.com/374655/149536775-f569f299-9fc0-44b7-9d76-eca41d573c3d.png)

## Possible Solutions

The effected elements do contain relatively "old" wikitext code. ... We have much better filter-logic at our hands with the latest TW version. eg: Instead of 

```
<$transclude field="caption">
  <$view field="title"/>
</$transclude>
```

To eliminate the problem we could write:

```
<$set name="field-name" filter="[all[current]get[caption]!is[blank]then[caption]else[title]]">
  <$view field=<<field-name>>/>
</$set>
```

The same / similar mechanism is used in several elements of the core-macros. ... 

At: https://github.com/Jermolene/TiddlyWiki5/pull/1191#issuecomment-65532894 @BramChen suggests: 

>Here is my solution: (working with the caption field)
>
>add a new template `$:/core/ui/ViewCaptionTemplate`
>
><pre><code>title: $:/core/ui/ViewCaptionTemplate
>
>\define tv-wikilinks() no
><$transclude field="caption"><$view field="title"/></$transclude>
></code></pre>
>
>
>    make changes in tiddlers (or source .tids):
>
>    `<$view field="title"/>` -> `{{||$:/core/ui/ViewCaptionTemplate}}`

## The Proposal

We should identify the core macros that contain the "empty caption" replaces the "title-link" problem and fix it in a way that would also give developers and translators the possibility to use a different field than "caption" eg: "title-fr" as discussed in PR #1191

@xcazin @Jermolene ... what do you think?

**Comment by Jermolene** [comment]
> To eliminate the problem we could write:

Not quite. We *transclude* the caption field (to allow for translations via further transclusions etc), but *view* the title field.

But it seems like the root of the problem is the unexpected behaviour of the transclude widget in treating a blank or empty field value as being present.

**Comment by linonetwo** [comment]
I'm using this in my plugin

```tid
<% if [all[current]has[caption]] %>
	<$transclude field="caption"/>
<% else %>
	<$view field="title"/>
<% endif %>
```

**Comment by Jermolene** [comment]
Hi @linonetwo it would be more usual to do this:

```
<$transclude field="caption">
	<$view field="title"/>
</$transclude>
```

**Comment by linonetwo** [comment]
@Jermolene I think this is what this issue talk about, `<$transclude field="caption">` show empty string when `caption` field exists but contains empty string!

**Comment by Jermolene** [comment]
Hi @Jermolene ah yes, if that's a constraint then the approach you suggest works well.
